plugins/csp-wms-overlays/src/WebMapTextureLoader.cpp

```diff
@@ -0,0 +1,358 @@
+////////////////////////////////////////////////////////////////////////////////////////////////////
+//                               This file is part of CosmoScout VR                               //
+//      and may be used under the terms of the MIT license. See the LICENSE file for details.     //
+//                        Copyright: (c) 2019 German Aerospace Center (DLR)                       //
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#include "WebMapTextureLoader.hpp"
+
+#include "WebMapException.hpp"
+
+#include "logger.hpp"
+
+#include "../../../src/cs-utils/convert.hpp"
+#include "../../../src/cs-utils/filesystem.hpp"
+#include "../../../src/cs-utils/utils.hpp"
+
+#include <boost/algorithm/string.hpp>
+#include <boost/filesystem.hpp>
+#include <boost/range/algorithm/replace_copy_if.hpp>
+#include <curlpp/Infos.hpp>
+#include <curlpp/Options.hpp>
+
+#define STB_IMAGE_IMPLEMENTATION
+#define STB_IMAGE_WRITE_IMPLEMENTATION
+#include <stb_image.h>
+#include <stb_image_write.h>
+
+namespace csp::wmsoverlays {
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+WebMapTextureLoader::WebMapTextureLoader()
+    : mThreadPool(32) {
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+WebMapTextureLoader::~WebMapTextureLoader() {
```

Comment:
You can remove empty destructors and just define them in the header with `= default`.